src/Type/grammar.pp

```diff
@@ -2,22 +2,28 @@
 
 %token parenthesis_ <
 %token _parenthesis >
+%token empty_string ""|''
+%token number        (\+|\-)?(0|[1-9]\d*)(\.\d+)?
```

Comment:
This won't allow floats in the following form: `.123`
Should be written as something similar:
```
[+-]?(?:(?:0|[1-9]\d*(?:\.0*[1-9]\d*)?)|\.0*\d*[1-9])
```
https://regex101.com/r/Nadose/1

Also this doesn't handle exponential floats.

Reply:
That's true. My goal was to solve the reported bug. A more complete implementation is welcome